core/src/components/checkbox/checkbox.scss

```diff
@@ -111,8 +111,12 @@
   display: none;
 }
 
+/**
+  * The native input must be hidden with display instead of visibility or 
+  * aria-hidden to avoid accessibility issues with nested interactive elements
+ */
```

Comment:
```suggestion
/**
 * The native input must be hidden with display instead of visibility or 
 * aria-hidden to avoid accessibility issues with nested interactive elements.
 */
```